pkgs/fc/qemu/clean-logs.py

```diff
@@ -0,0 +1,67 @@
+"""
+A helper to clean up log files from fc.qemu.
+
+The logrotate targets aren't smart enough because we have VMs that move around
+and thus would end up with missing files, but `missingok` doesn't help if the
+pattern doesn't match any more.
+
+In addition we may have files that have not been written to for a long time but
+do correspond to a VM still running here.
+
+So this script deletes log files, iff:
+
+1. the file is not open, AND
+2. it is older than 14 days.
+
+"""
+import datetime
+import pathlib
+import re
+import subprocess
+import time
+
+MAX_AGE = datetime.timedelta(days=14)
+VM_LOG_DIR = pathlib.Path("/var/log/vm")
+VM_LOG_NAME_PATTERN = re.compile(r"[0-9a-zA-Z]+\.log")
+
+# Find logs of running VMs
+
+print("Checking open files ...")
+
+lsof = subprocess.run(
+    ["lsof", "-F", "n", "+D", VM_LOG_DIR], capture_output=True
+)
+# Expected output:
+# $ lsof +D /var/log/vm -F n
+# p23052
+# n/var/log/vm/release2305dev01.log
+# p319664
+# n/var/log/vm/mailstubtest01.log
+# p321165
+# n/var/log/vm/services34.log
+# p332886
+# n/var/log/vm/services08.log
+# p333454
+
+open_files = set()
+
+for open_file in lsof.stdout.decode("ascii").splitlines():
```

Comment:
In case we ever get non-ascii VMs, this might break. An `errors="ignore"` makes this more resilient.

Also, encoding could already be specifid to `subprocess.run`, but I guess that's a matter of taste.

Reply:
I did that on purpose. Ignoring errors here means that we won't properly match filenames. I decided that we really do not expect non-ASCII to appear here due to other impacts in the chain.